src/EntityRevisionTrait.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\entity\EntityRevisionTrait.
+ */
+
+namespace Drupal\entity;
+
+/**
+ * Provides fixes for revisions
+ */
+trait EntityRevisionTrait {
```

Comment:
Other than the docblocks all needing rewrites, this trait doesn't make a lot of sense to me.
The base entity class already has public methods getEntityType, getEntityTypeId, getRevisionId(), how are we defining them as protected here?

I think revisions are a fine exception to the "don't provide a base class" rule, especially since we're attempting to use a trait to fix a method of a parent base class.